src/asfvideo.cpp

```diff
@@ -0,0 +1,770 @@
+#endif
+        io_->seek(io_->tell() + nameLength, BasicIo::beg);
+      } else {
+        io_->read(buf.data(), nameLength);
+      }
+
+      v->read(toString16(buf));
+      if (dataType == 6) {
+        io_->read(guidBuf, 16);
+        getGUID(guidBuf, fileID);
+      } else
+          // Sanity check with an "unreasonably" large number
+          if (dataLength > 5000) {
+#ifndef SUPPRESS_WARNINGS
+        EXV_ERROR << "Xmp.video.Metadata dataLength was found to be larger than 5000 "
+                  << " entries considered invalid; not read.\n";
+#endif
+        io_->seek(io_->tell() + dataLength, BasicIo::beg);
+      } else
+        io_->read(buf.data(), dataLength);
+    }
+
+    else if (meta == 2) {
+      io_->read(buf.data(), 2);
+      nameLength = Exiv2::getUShort(buf.data(), littleEndian);
+
+      if (nameLength > 5000) {
+#ifndef SUPPRESS_WARNINGS
+        EXV_ERROR << "Xmp.video.Metadata nameLength was found to be larger than 5000 "
+                  << " entries considered invalid; not read.\n";
+#endif
+        io_->seek(io_->tell() + nameLength, BasicIo::beg);
+      } else {
+        io_->read(buf.data(), nameLength);
+      }
+
+      v->read(toString16(buf));
+
+      io_->read(buf.data(), 2);
+      dataType = Exiv2::getUShort(buf.data(), littleEndian);
+
+      io_->read(buf.data(), 2);
+      dataLength = Exiv2::getUShort(buf.data(), littleEndian);
+
+      // Sanity check with an "unreasonably" large number
+      if (dataLength > 5000) {
+#ifndef SUPPRESS_WARNINGS
+        EXV_ERROR << "Xmp.video.Metadata dataLength was found to be larger than 5000 "
+                  << " entries considered invalid; not read.\n";
+#endif
+        io_->seek(io_->tell() + dataLength, BasicIo::beg);
+      } else
+        io_->read(buf.data(), dataLength);
+    }
+
+    if (dataType == 0) {  // Unicode String
+      v->read(toString16(buf));
+    } else if (dataType == 2 || dataType == 5) {  // 16-bit Unsigned Integer
+      v->read(Exiv2::toString(Exiv2::getUShort(buf.data(), littleEndian)));
+    } else if (dataType == 3) {  // 32-bit Unsigned Integer
+      v->read(Exiv2::toString(Exiv2::getULong(buf.data(), littleEndian)));
+    } else if (dataType == 4) {  // 64-bit Unsigned Integer
+      v->read(Exiv2::toString(getUint64_t(buf)));
+    } else if (dataType == 6) {  // 128-bit GUID
+      v->read(Exiv2::toString(fileID));
+    } else {  // Byte array
+      v->read(Exiv2::toString(buf.data()));
+    }
+  }
+
+  if (meta == 1) {
+    xmpData().add(Exiv2::XmpKey("Xmp.video.Metadata"), v.get());
+  } else if (meta == 2) {
+    xmpData().add(Exiv2::XmpKey("Xmp.video.ExtendedContentDescription"), v.get());
+  } else {
+    xmpData().add(Exiv2::XmpKey("Xmp.video.MetadataLibrary"), v.get());
+  }
+}  // AsfVideo::metadataHandler
+
+void AsfVideo::fileProperties() {
+  DataBuf buf(9);
+  buf.data()[8] = '\0';
+
+  byte guidBuf[16];
+  io_->read(guidBuf, 16);
+  char fileID[37] = "";
```

Comment:
is it really needed to initialize the buffer with `""` ? 
[suggestion] I would replace the magic value `37` with a constant variable with a good name. I noticed that we are using that magic value in several places.

Reply:
Done here ad72e1e6b0c2a10383a3878c1e603aa78fe3275a